creditcoin-js/src/extrinsics/add-ask-order.ts

```diff
@@ -40,9 +40,8 @@ export const addAskOrderAsync = async (
     signer: KeyringPair,
 ) => {
     return new Promise<AskOrderAdded>((resolve, reject) => {
-        const onFail = (result: SubmittableResult) => reject(handleTransactionFailed(api, result));
         const onSuccess = (result: SubmittableResult) => resolve(processAskOrderAdded(api, result));
-        addAskOrder(api, lenderAddressId, loanTerms, expirationBlock, guid, signer, onSuccess, onFail).catch((reason) =>
+        addAskOrder(api, lenderAddressId, loanTerms, expirationBlock, guid, signer, onSuccess, reject).catch((reason) =>
```

Comment:
I don't understand what's the scope of the original request, CSUB-282 nor why this needs to be removed. 

I also see that `expectNoDispatchError` has been moved into creditcoin-js but not actually used internally. If this isn't going to be used by creditcoin-js then why not leave it inside integration-tests/ like it was?

Reply:
It is used here(https://github.dev/gluwa/creditcoin/blob/099cc7169d8322071fa8d76b388f0adc995a5dc4/creditcoin-js/src/extrinsics/common.ts#L17).

Reply:
I updated the description. Those changes are a byproduct of trying to unify error handling and getting rid of callbacks as much as possible.